v3/src/components/data-display/models/data-display-layer-union.ts

```diff
@@ -0,0 +1,40 @@
+import {Instance, types} from "mobx-state-tree"
+import {DataDisplayLayerModel, IDataDisplayLayerModel, kUnknownLayerModelType} from "./data-display-layer-model"
+import {GraphPointLayerModel, IGraphPointLayerModel, IGraphPointLayerModelSnapshot}
+  from "../../graph/models/graph-point-layer-model"
+import {IMapPointLayerModel, IMapPointLayerModelSnapshot, MapPointLayerModel}
+  from "../../map/models/map-point-layer-model"
+import {MapLayerModel} from "../../map/models/map-layer-model"
+import {IMapBaseLayerModel, IMapBaseLayerModelSnapshot, MapBaseLayerModel} from "../../map/models/map-base-layer-model"
+import {IMapPolygonLayerModel, IMapPolygonLayerModelSnapshot, MapPolygonLayerModel}
+  from "../../map/models/map-polygon-layer-model"
+
+export const UnknownDataDisplayLayerModel = DataDisplayLayerModel
+  .named("UnknownDataDisplayLayerModel")
+  .props({
+    type: "Unknown"
+  })
+export interface IUnknownDataDisplayLayerModel extends Instance<typeof UnknownDataDisplayLayerModel> {}
+
+export function isUnknownDataDisplayLayerModel(
+  dataDisplayModel: IDataDisplayLayerModel): dataDisplayModel is IUnknownDataDisplayLayerModel {
+  return dataDisplayModel.type === kUnknownLayerModelType
+}
+
+
+const dataDisplayLayerTypeDispatcher = (displayLayerModelSnap: IDataDisplayLayerSnapshotUnion) => {
+  switch (displayLayerModelSnap.type) {
+    case "graphPointLayer": return GraphPointLayerModel
+    case "mapBaseLayer": return MapBaseLayerModel
+    case "mapPointLayer": return MapPointLayerModel
+    case "mapPolygonLayer": return MapPolygonLayerModel
+    default: return MapLayerModel
```

Comment:
Now that we have an `UnknownDataDisplayLayerModel`, this would be the place to return it.

Reply:
Oops.